map-of-pi/map-of-pi-backend-react
# Initialize backend for shared server hosting.

Merged some changes from @soleil00's ft-auth branch.

**What This PR Does**
Implemented basic user authentication functionality. This allows users to authenticate from the frontend to the backend.

**How to Test This**
1. Reference utils/env.ts to build a .env file and populate the following variables accordingly ->
    - JWT_SECRET = MAP_OF_PI
    - PLATFORM_API_URL = https://api.minepi.com
    - PI_API_KEY =  **INSERT YOUR PI API KEY**
    - MONGODB_URL = mongodb://localhost:27017/demoDB
    - PORT=8001
    - CLOUDINARY_CLOUD_NAME = **INSERT YOUR CLOUDINARY NAME**
    - CLOUDINARY_API_KEY = **INSERT YOUR CLOUDINARY API KEY**
    - CLOUDINARY_API_SECRET = **INSERT YOUR CLOUDINARY_API_SECRET**
    - DEVELOPMENT_URL = http://localhost:8001

2. Start the development server by running `npm run dev`.
3. Ensure your frontend is running in sandbox mode.
4. Adjust the port settings as needed for both the frontend and backend.
5. Once the application is running in the sandbox environment, you will encounter an authentication screen from the PI SDK.
6. Check the backend terminal logs to verify that user data is being logged correctly.

**[see commits]**
